### ISSUE TITLE
Autostart VMs

### ISSUE THREAD

**Issue by juk81** [hyperbox/hyperbox#5]
We need to make the setting automatic start virtual machines

**Comment by mribichich** [comment]
+1

**Comment by mribichich** [comment]
Hi, is there any way to accomplish this until the feature is ready? thanks!

**Comment by maxidor** [comment]
Yes, you can use the [VirtualBox Autostart feature](http://www.virtualbox.org/manual/ch09.html#autostart) under the same user as the Hyperbox process.

This feature, while being simple in description, becomes quite complicated when it comes to make it nice and polished. I'm currently working on some implementations and should release *something* in the coming weeks.

This is actually getting more and more a priority internally too as we really start to miss this feature in-house.
Stay tuned!

**Comment by marcosx86** [comment]
Hi @maxidor any news about auto-start machines? I could help if you wish so, have some ideas. Also used a set of scripts of mine to start those with phpvirtualbox (see my repo /vboxdepstart), but I'm testing other solutions, including yours.

**Comment by maxidor** [comment]
@marcosx86 I just finished migrating the whole project to Gradle build, which was one of my requirements - and a dire lack of time until now made this much longer than I wanted.
I'll resume working on this very soon...

**Comment by marcosx86** [comment]
Nice to see, you spawn VB processes under VBoxSVC over shell, not by spawning under the hboxd, so if it crashes, you does not lose the VMs ; )

Also, I can auto start my machines using my old behavior, while hboxd doesn't have it built in.

**Comment by maxidor** [comment]
@marcosx86 if you use the `XPCOM` connector, yes I do. If you use the `WebSercvices` connector, `hboxd` start its own instance of `vboxwebsrv` which is the real client from `VBoxSVC` point of view.

**Comment by marcosx86** [comment]
I'm using default setup by master branch, just gradlew'd everything and it's running smoothly. `HTOP` says `VboxSVC` is detached from `vboxwebsrv`, and this one, yes, it's child of `hboxd`.

**Comment by maxidor** [comment]
Glad to hear it works good for you!

**Comment by krselander** [comment]
my virtual box will not work at all. What do i do

**Comment by maxidorius** [comment]
@krselander Please open a new issue and describe the exact steps you are taking and your exact problem (error message, stacktrace, versions of hyperbox and virtualbox, connector used, etc.)
